relayer/relays/parachain/catchup.go

```diff
@@ -189,22 +189,23 @@ func (li *BeefyListener) parablocksWithProofs(blocks []ParaBlockWithDigest, late
 
 		log.Debug("Have inputs for proof generation")
 		for i, v := range allParaHeads {
-			log.WithFields(logrus.Fields{
-				"LeafIndex": i,
-				"HeadData": fmt.Sprintf("%#x", v),
+			log.WithFields(log.Fields{
+				"LeafIndex":        i,
+				"HeadData":         fmt.Sprintf("%#x", v),
 				"IsSnowbridgePara": i == ownParaHeadLeafIndex,
 			}).Debug("Head data")
 		}
 
-		// Note - relayChainBlockNumber will be one less than the block number discovered
+		// Note - relayChainBlockNumber will be one less than the actual block number we want
 		// due to the decrement at the end of the loop, but the mmr leaves are 0 indexed whereas
 		// block numbers start from 1, so we actually do want to query for the leaf at
-		// one less than the block number
-		mmrProof, err := li.relaychainConn.GetMMRLeafForBlock(uint64(relayChainBlockNumber), relayBlockHash)
+		// relayChainBlockNumber
+		mmrProof, err := li.relaychainConn.GetMMRLeafForBlock(uint64(relayChainBlockNumber), latestRelaychainBlockHash)
```

Comment:
fyi: this is the critical change - querying for proofs to the same latestRelaychainBlockHash, irrespective of whether the proof is for a new or older relayChainBlockNumber